crates/ty/src/lib.rs

```diff
@@ -314,14 +320,19 @@ impl MainLoop {
                             let mut max_severity = Severity::Info;
                             let diagnostics_count = result.len();
 
+                            let mut stdout = self.printer.stream_for_details().lock();
                             for diagnostic in result {
-                                write!(stdout, "{}", diagnostic.display(db, &display_config))?;
+                                // Only render diagnostics if they're going to be displayed, since doing
+                                // so is expensive.
+                                if stdout.is_enabled() {
```

Comment:
We should move this check outside the `for` loop as it won't change for all diagnostics.

Reply:
Yep haha, that broke the tests and I was like.. oh mhm